integration/tests/basic/evm/test_solana_interoperability.py

```diff
@@ -41,6 +43,61 @@ class TestSolanaInteroperability:
     accounts: EthAccounts
     web3_client: NeonChainWeb3Client
 
+    @pytest.fixture(scope="class")
+    def account_sol_network(self, class_account_sol_chain):
+        return class_account_sol_chain
+
+    @pytest.fixture(scope="class")
+    def call_solana_caller_sol_network(self, account_sol_network, web3_client_sol):
+        contract, _ = web3_client_sol.deploy_and_get_contract(
+            contract="precompiled/CallSolanaCaller.sol",
+            version="0.8.10",
+            contract_name="CallSolanaCaller",
+            account=account_sol_network,
+        )
+        return contract
+
+    def get_transfer_instruction(self, sol_client, from_wallet, to_wallet, amount, contract, is_set_authority=True):
+        mint = spl.token.client.Token.create_mint(
+            conn=sol_client,
+            payer=from_wallet,
+            mint_authority=from_wallet.pubkey(),
+            decimals=9,
+            program_id=TOKEN_PROGRAM_ID,
+        )
+        mint.payer = from_wallet
+        from_token_account = mint.create_associated_token_account(from_wallet.pubkey())
+        to_token_account = mint.create_associated_token_account(to_wallet.pubkey())
+        mint.mint_to(
+            dest=from_token_account,
+            mint_authority=from_wallet,
+            amount=amount,
+            opts=TxOpts(skip_confirmation=False, skip_preflight=True),
+        )
+
+        authority_pubkey: bytes = contract.functions.getSolanaPDA(bytes(TRANSFER_TOKENS_ID), b"authority").call()
+        if is_set_authority:
+            mint.set_authority(
+                from_token_account,
+                from_wallet,
+                spl.token.instructions.AuthorityType.ACCOUNT_OWNER,
+                Pubkey(authority_pubkey),
+                opts=TxOpts(skip_confirmation=False, skip_preflight=True),
+            )
+
+        instruction = Instruction(
+            program_id=TRANSFER_TOKENS_ID,
+            accounts=[
+                AccountMeta(from_token_account, is_signer=False, is_writable=True),
+                AccountMeta(mint.pubkey, is_signer=False, is_writable=True),
+                AccountMeta(to_token_account, is_signer=False, is_writable=True),
+                AccountMeta(Pubkey(authority_pubkey), is_signer=False, is_writable=True),
+                AccountMeta(TOKEN_PROGRAM_ID, is_signer=False, is_writable=False),
+            ],
+            data=bytes([0x0]),
+        )
+        return serialize_instruction(TRANSFER_TOKENS_ID, instruction), mint, [from_token_account, to_token_account]
```

Comment:
It is very difficult to read when some function return more then 1 value
You can leave it as it is if it is the only option :)